src/benchmarking.rs

```diff
@@ -16,21 +16,640 @@
+
+	fund_feed{
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
+		let query_id = Keccak256::hash(query_data.as_ref()).into();
+		let feed_creator = account::<AccountIdOf<T>>("account", 1, SEED);
+
+		T::BenchmarkHelper::set_balance(feed_creator.clone(), 1000);
+		let feed_id = create_feed::<T>(feed_creator.clone(),
+				query_id,
+				token::<T>(10u64),
+				T::Time::now().as_secs(),
+				700,
+				60,
+				0,
+				token::<T>(0u64),
+				query_data.clone(),
+				token::<T>(1000u64)
+		);
+
+	}: _(RawOrigin::Signed(feed_creator.clone()), feed_id, query_id, token::<T>(10u64))
+	verify {
+		assert!(<DataFeeds<T>>::get(query_id, feed_id).is_some());
+	}
+
+	submit_value {
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
+		let query_id = Keccak256::hash(query_data.as_ref()).into();
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let address = Address::zero();
+		// report deposit stake
+		deposit_stake::<T>(reporter.clone(), trb(1200), address)?;
+        T::BenchmarkHelper::set_time(HOURS);
+	}: _(RawOrigin::Signed(reporter.clone()), query_id, uint_value::<T>(4_000), 0, query_data.clone())
+	verify {
+		assert!(<StakerDetails<T>>::get(reporter).is_some());
+	}
+
+	add_staking_rewards {
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		T::BenchmarkHelper::set_balance(reporter.clone(), 1000);
+	}: _(RawOrigin::Signed(reporter), token::<T>(100u64))
+
+	update_stake_amount {
+		let staking_token_price_query_data: QueryDataOf<T> =
+			spot_price("trb", "usd").try_into().unwrap();
+		let staking_token_price_query_id = Keccak256::hash(staking_token_price_query_data.as_ref()).into();
+		let staking_to_local_token_query_data: QueryDataOf<T> =
+			spot_price("trb", "ocp").try_into().unwrap();
+		let staking_to_local_token_query_id: QueryId =
+			Keccak256::hash(staking_to_local_token_query_data.as_ref()).into();
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let address = Address::zero();
+		// report deposit stake
+		deposit_stake::<T>(reporter.clone(), trb(10_000), address)?;
+		T::BenchmarkHelper::set_time(HOURS);
+		// submit value
+		Tellor::<T>::submit_value(
+			RawOrigin::Signed(reporter.clone()).into(),
+			staking_token_price_query_id,
+			uint_value::<T>(50 * 10u128.pow(18)),
+			0,
+			staking_token_price_query_data.clone())?;
+		T::BenchmarkHelper::set_time(12 * HOURS);
+
+		Tellor::<T>::submit_value(
+			RawOrigin::Signed(reporter.clone()).into(),
+			staking_to_local_token_query_id,
+			uint_value::<T>(6 * 10u128.pow(18)),
+			0,
+			staking_to_local_token_query_data)?;
+
+		T::BenchmarkHelper::set_time(12 * HOURS);
+	}: _(RawOrigin::Signed(reporter))
+
+	tip {
+		let s in 2..T::MaxTimestamps::get();
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
```

Comment:
It seems `query_data` is defined and used a quite a few times in loops in this file for benchmarking, so you might consider varying the query data to be larger and smaller than just spot prices. Some other types [here](https://github.com/tellor-io/dataSpecs/tree/main/types), or you could just generate them randomly within bounds of query data max size.

Reply:
Thanks for the suggestion, will resolve this comment in #94, as this PR is specific to sending weights to parachain contract.

Will clean this PR after merging #94 into main.

Reply:
Sounds like querydata should always be max length based on chain config to generate the worst case weight and then benchmark should take in actual query data length so caller can be refunded at runtime.